### ISSUE TITLE
multi-GPU support?

### ISSUE THREAD

**Issue by zhenxingjian** [scverse/scvi-tools#2355]
I would like to ask if we're supporting multi-GPU training now?
I'm testing the scvi version 1.0.4.
Shall I try 1.1.x?

**Comment by zhenxingjian** [comment]
I've changed the devices=-1 (from lightning.pytorch, this means to use all GPU)
I got the following messages:

2023-12-07 12:20:22.818922 -- 0
An NVIDIA GPU may be present on this machine, but a CUDA-enabled jaxlib is not installed. Falling back to cpu.
GPU available: True (cuda), used: True
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs
Initializing distributed: GLOBAL_RANK: 0, MEMBER: 1/2
/miniconda3/envs/scvi_latest/lib/python3.10/site-packages/scvi/_settings.py:63: UserWarning: Since v1.0.0, scvi-tools no longer uses a random seed by default. Run `scvi.settings.seed = 0` to reproduce results from previous versions.
  self.seed = seed
/miniconda3/envs/scvi_latest/lib/python3.10/site-packages/scvi/_settings.py:70: UserWarning: Setting `dl_pin_memory_gpu_training` is deprecated in v1.0 and will be removed in v1.1. Please pass in `pin_memory` to the data loaders instead.
  self.dl_pin_memory_gpu_training = (
2023-12-07 12:21:15.014003 -- 0
An NVIDIA GPU may be present on this machine, but a CUDA-enabled jaxlib is not installed. Falling back to cpu.
Initializing distributed: GLOBAL_RANK: 1, MEMBER: 2/2
----------------------------------------------------------------------------------------------------
distributed_backend=nccl
All distributed processes registered. Starting with 2 processes
----------------------------------------------------------------------------------------------------

You are using a CUDA device ('NVIDIA A100-PCIE-40GB') that has Tensor Cores. To properly utilize them, you should set `torch.set_float32_matmul_precision('medium' | 'high')` which will trade-off precision for performance. For more details, read https://pytorch.org/docs/stable/generated/torch.set_float32_matmul_precision.html#torch.set_float32_matmul_precision

[ Note, I'm using PyTorch not Jax, so the line of jaxlib should be irrelevant ]

However, it's not training at all. All the GPU shows 100% util, but not actual training. the Pwr Usg is very low (60W/ 250W)
And the training process got stuck.

Do you have any idea about why?

**Comment by zhenxingjian** [comment]
> Hi, multi-GPU training is not available in 1.0.4 but will be in 1.1.0, so please hold tight! We'll be releasing a release candidate soon for it. Feel free to also install from main and try it out.

Hi Martin,

I've tried the scvi==1.1.0 (built from main), but I got the same error.
All GPUs shows 100% GPU-Util but not training at all.

2023-12-08 11:07:59.989550 -- 0
GPU available: True (cuda), used: True
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs
Initializing distributed: GLOBAL_RANK: 0, MEMBER: 1/2
2023-12-08 11:08:42.294553 -- 0
Initializing distributed: GLOBAL_RANK: 1, MEMBER: 2/2
----------------------------------------------------------------------------------------------------
distributed_backend=nccl
All distributed processes registered. Starting with 2 processes
----------------------------------------------------------------------------------------------------

You are using a CUDA device ('NVIDIA A100-PCIE-40GB') that has Tensor Cores. To properly utilize them, you should set `torch.set_float32_matmul_precision('medium' | 'high')` which will trade-off precision for performance. For more details, read https://pytorch.org/docs/stable/generated/torch.set_float32_matmul_precision.html#torch.set_float32_matmul_precision
